contrib/README.md

```diff
@@ -0,0 +1,324 @@
+# Contributions
+
+This package contains sub-packages that are contributions from students.
+
+Teaching Assistants reviewed the code before merging, but are not responsible for it.
+
+If you think you also have a piece of code that is worth adding to this project,
+please do not hesitate to reach out (via a pull request or by contacting a teaching assistant)!
+
+## List of contributions
+
+- [Contribution 1 - UART Reader](#uart-reader)
+  - [UART Reader Manual](#uart-reader-in-depth-manual)
```

Comment:
```suggestion
**List of contributions:**

- [UART Reader](#uart-reader)
  - [Manual](#uart-reader-in-depth-manual)
```

I'd avoid the extra wording and the unnecessary heading.

The documentation seems very long. I'd keep it short, and maybe create a second markdown file, `UART_READER.md`, that contains more details.